src/other/Checkbox.js

```diff
@@ -0,0 +1,95 @@
+    Checkbox.prototype.publish("valueColumn", null, "set", "Checkbox display value", function () { return this.columns(); }, { optional: true });
+    Checkbox.prototype.publish("textColumn", null, "set", "Checkbox value(s)", function () { return this.columns(); }, { optional: true });
+    Checkbox.prototype.publish("sort", null, "set", "Sort contents", ["", "ascending", "descending"], { optional: true });
+
+    Checkbox.prototype.checkboxData = function () {
+        var view = this._db.rollupView([this.textColumn(), this.valueColumn()]);
+        var retVal = [];
+        retVal = retVal.concat(view.entries().map(function (row) {
+            return {
+                text: row.key,
+                value: row.values.length ? row.values[0].key : "",
+                origRow: row.values.length && row.values[0].values.length ? row.values[0].values[0] : []
+            };
+        }, this));
+        if (this.sort_exists()) {
+            var descending = this.sort() === "descending";
+            retVal.sort(function (l, r) {
+                if (l.text < r.text) return descending ? 1 : -1;
+                if (l.text > r.text) return descending ? -1 : 1;
+                return 0;
+            });
+        }
+
+        return retVal;
+    };
+
+    Checkbox.prototype.enter = function (domNode, element) {
+        HTMLWidget.prototype.enter.apply(this, arguments);
+        this._span = element.append("span");
+        this._label = this._span.append("label")
+            .attr("for", this.id() + "_select")
+        ;
+
+        this._Checkbox = this._span.append("div")
+            .attr("id", this.id() + "_select");
+    };
+
+    Checkbox.prototype.update = function (domNode, element) {
+        HTMLWidget.prototype.update.apply(this, arguments);
+        var context = this;
+        this._label
+            .text(this.label());
+        var CheckboxCont_Div = this._Checkbox.selectAll(".dataRow").data(this.checkboxData());
+        var label = CheckboxCont_Div.enter().append("label");
+        label.attr("class", "dataRow");
+        label.text(function(row){
+            return row.text;
+        }).on("change", function (selectedData) {
+            var options = [];
+            if (selectedData && selectedData.origRow) {
+                options.push(selectedData.origRow);
+            }
+
+            if (options.length) {
+                context.click(context.rowToObj(options[0]), context.valueColumn(), true); //TODO:  Multiselect not support in HIPIE
+            } else {
+                context.click([], context.valueColumn(), false);
+            }
+        });
+        var  Checkbox = label.append('input');
+            Checkbox.attr("type","Checkbox")
+                .attr("name", "tableCheckboxGroup")
+                .attr("value", function (row) { return row.value; });
+
+        CheckboxCont_Div.exit().remove();
+        CheckboxCont_Div.order();
```

Comment:
This enter/update/exit code looks wrong - the "label.text" is only set on enter for example.